pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,125 @@
-
-        remote_api = RemotesContainerApi(self.client_api)
-        remote = remote_api.create(gen_container_remote())
-        self.addCleanup(remote_api.delete, remote.pulp_href)
-
-        self.assertEqual(repository.latest_version_href, f"{repository.pulp_href}versions/0/")
-        repository_sync_data = ContainerRepositorySyncURL(remote=remote.pulp_href)
-
-        # Sync the repository.
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
-        repository = repository_api.read(repository.pulp_href)
-        self.assertIsNotNone(repository.latest_version_href)
-
-        # Sync the repository again.
-        latest_version_href = repository.latest_version_href
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
-        repository = repository_api.read(repository.pulp_href)
-        self.assertEqual(latest_version_href, repository.latest_version_href)
-
-
-class SyncInvalidURLTestCase(PulpTestCase):
-    """Sync a repository with an invalid url on the Remote."""
+import pytest
+from pulpcore.tests.functional import PulpTaskError
 
-    def test_all(self):
-        """
-        Sync a repository using a Remote url that does not exist.
-
-        Test that we get a task failure.
-
-        """
-        client_api = gen_container_client()
-
-        repository_api = RepositoriesContainerApi(client_api)
-        repository = repository_api.create(ContainerContainerRepository(**gen_repo()))
-        self.addCleanup(repository_api.delete, repository.pulp_href)
-
-        remote_api = RemotesContainerApi(client_api)
-        remote_data = gen_container_remote(url="http://i-am-an-invalid-url.com/invalid/")
-        remote = remote_api.create(ContainerContainerRemote(**remote_data))
-        self.addCleanup(remote_api.delete, remote.pulp_href)
-
-        repository_sync_data = ContainerRepositorySyncURL(remote=remote.pulp_href)
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-
-        with self.assertRaises(PulpTaskError) as context:
-            monitor_task(sync_response.task)
-        self.assertIsNotNone(context.exception.task.error["description"])
+from pulp_container.tests.functional.utils import gen_container_remote
+from pulp_container.tests.functional.constants import PULP_FIXTURE_1
 
+from pulpcore.client.pulp_container import ContainerContainerRemote
 
-class TestRepeatedSync(PulpTestCase):
-    """Test behavior when a sync is repeated."""
 
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        cls.client_api = gen_container_client()
+@pytest.fixture
+def synced_repository_with_filtered_tags(
+    container_repository_factory, container_remote_factory, sync
+):
+    def _synced_repository_with_filtered_tags(include_tags=None, exclude_tags=None):
```

Comment:
How about:
```suggestion
    def _synced_repository_with_filtered_tags(url=None, upstream_name=None, include_tags=None, exclude_tags=None):
```
It would make the fixture most versatile i think. Maybe even name it `synced_container_repository_factory`.